src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs

```diff
@@ -48,7 +48,7 @@ internal sealed class ExceptionHandlerMiddlewareImpl
         _metrics = new DiagnosticsMetrics(meterFactory);
         _problemDetailsService = problemDetailsService;
 
-        if (_options.ExceptionHandler == null)
+        if (_options.ExceptionHandler == null && _exceptionHandlers.Length == 0)
```

Comment:
Uhmm, Is this a breaking change!? I might want to handle some exceptions with `IExceptionHandler` and if they couldn't handle them, I want to execute the pipeline again. (`_options.ExceptionHandler = _next;`)

Reply:
I don't know whether it was intended to work that way, but it certainly appears to, so changing it feels wrong.

Reply:
> I might want to handle some exceptions with IExceptionHandler and if they couldn't handle them, I want to execute the pipeline again.

Is this supported at the moment (sans this change)?

Reply:
There's a bit of a mess right now where an exception could be handled in one of three ways, an exception handler, problem details, or a delegate (e.g. re-execute). I tried to reconcile problem details and exception handlers but it was too breaking.